core/domain/draft_upgrade_services.py

```diff
@@ -88,6 +90,135 @@ def try_upgrading_draft_to_exp_version(
 class DraftUpgradeUtil(python_utils.OBJECT):
     """Wrapper class that contains util functions to upgrade drafts."""
 
+    @classmethod
+    def _convert_states_v33_dict_to_v34_dict(cls, draft_change_list):
+        """Converts draft change list from state version 33 to 34. State
+        version 34 adds the new schema for Math RTEs.
+
+        Args:
+            draft_change_list: list(ExplorationChange). The list of
+                ExplorationChange domain objects to upgrade.
+
+        Returns:
+            list(ExplorationChange). The converted draft_change_list.
+        """
+        for i, change in enumerate(draft_change_list):
+            if (change.cmd == exp_domain.CMD_EDIT_STATE_PROPERTY and
+                    change.property_name ==
+                    exp_domain.STATE_PROPERTY_INTERACTION_ANSWER_GROUPS):
+                converted_new_value = (
+                    [(
+                        state_domain.AnswerGroup.convert_html_in_answer_groups(
+                            answer_group,
+                            html_validation_service.
+                            add_math_content_to_math_rte_components))
+                     for answer_group in change.new_value])
+                draft_change_list[i] = exp_domain.ExplorationChange({
+                    'cmd': exp_domain.CMD_EDIT_STATE_PROPERTY,
+                    'property_name': (
+                        exp_domain.STATE_PROPERTY_INTERACTION_ANSWER_GROUPS),
+                    'state_name': change.state_name,
+                    'new_value': converted_new_value
+                })
+            elif (change.cmd == exp_domain.CMD_EDIT_STATE_PROPERTY and
+                  change.property_name ==
+                  exp_domain.STATE_PROPERTY_INTERACTION_CUST_ARGS):
+                if 'choices' in change.new_value.keys():
+                    for value_index, value in enumerate(
+                            change.new_value['choices']['value']):
```

Comment:
What if the new_value doesn't have `choices` key?

Reply:
basically, we check for the `choices` key in the customization arg in order to find whether the interaction is one of `ItemSelection`, `MultichoiceInput`, or `DragAndDrop`.
If `choices` is not among keys, then we don't have to convert the `change.new_value`
This approach is already been used in the codebase here  [link](https://github.com/oppia/oppia/blob/a4a3a7266858efb54ae7ceba3a7c834837744911/core/domain/draft_upgrade_services.py#L108)